spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -276,8 +297,10 @@ public void startAckThread() {
 	}
 
 	protected void send(DatagramPacket packet) throws Exception {
-		DatagramSocket socket = this.getSocket();
-		packet.setSocketAddress(this.getDestinationAddress());
+		DatagramSocket socket = getSocket();
+		if (this.socketExpression == null) {
+			packet.setSocketAddress(getDestinationAddress());
```

Comment:
OK. I see in the `DualStackPlainDatagramSocketImpl`:
```
if (p.getAddress() == null ||p.getData() ==null)
            throw new NullPointerException("null address || null buffer");
```
So, we always have to have `destinationAddress` in the `DatagramPacket` independently of socket...

Need to think more, but looks like the general idea behind `socket-expression` doesn't fit the reality...